resources/persist/src/lib.rs

```diff
@@ -111,13 +152,53 @@ mod tests {
         assert_eq!(result, "test");
     }
 
+    #[test]
+    fn test_list() {
+        let persist = PersistInstance {
+            service_name: ServiceName::from_str("test_list").unwrap(),
+        };
+
+        persist.save("test_list", "test_list").unwrap();
+
+        let result = vec!["shuttle_persist/test_list/test_list".to_string()];
+        let list_result = persist.list().unwrap();
+        assert_eq!(result, list_result);
+    }
+
+    #[test]
+    fn test_clear() {
+        let persist = PersistInstance {
+            service_name: ServiceName::from_str("test_clear").unwrap(),
+        };
+
+        persist.save("test_clear", "test_clear").unwrap();
+        persist.clear().unwrap();
+        let clear_result = persist.list().unwrap_err();
+        assert_eq!(
+            clear_result.to_string(),
+            "failed to list contents of folder: No such file or directory (os error 2)"
+        );
```

Comment:
Maybe it is unintuitive that calling list() on an empty persist store returns an error. If you also call `create_dir_all` in list() like save() does, you can make sure that the folder always exists when asked for, and an empty Vec can be returned.

Reply:
Now just removing the files instead of the entire folder, test adjusted accordingly.